src/core/src/package_managers/YumPackageManager.py

```diff
@@ -345,28 +370,434 @@ def get_package_size(self, output):
+
+        self.composite_logger.log("OS patch state per auto OS update service: [yum-cron={0}] [dnf-automatic={1}] [packagekit={2}]"
+                                  .format(str(current_auto_os_patch_state_for_yum_cron), str(current_auto_os_patch_state_for_dnf_automatic), str(current_auto_os_patch_state_for_packagekit)))
+
+        if current_auto_os_patch_state_for_yum_cron == Constants.AutomaticOSPatchStates.ENABLED \
+                or current_auto_os_patch_state_for_dnf_automatic == Constants.AutomaticOSPatchStates.ENABLED \
+                or current_auto_os_patch_state_for_packagekit == Constants.AutomaticOSPatchStates.ENABLED:
+            current_auto_os_patch_state = Constants.AutomaticOSPatchStates.ENABLED
+        elif current_auto_os_patch_state_for_yum_cron == Constants.AutomaticOSPatchStates.DISABLED \
+                and current_auto_os_patch_state_for_dnf_automatic == Constants.AutomaticOSPatchStates.DISABLED \
+                and current_auto_os_patch_state_for_packagekit == Constants.AutomaticOSPatchStates.DISABLED:
+            current_auto_os_patch_state = Constants.AutomaticOSPatchStates.DISABLED
+        else:
+            current_auto_os_patch_state = Constants.AutomaticOSPatchStates.UNKNOWN
 
-    def disable_auto_os_update(self):
-        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
-        # NOTE: Implementation pending
-        pass
+        self.composite_logger.log_debug("Overall Auto OS Patch State based on all auto OS update service states [OverallAutoOSPatchState={0}]".format(str(current_auto_os_patch_state)))
+        return current_auto_os_patch_state
 
-    def is_image_default_patch_configuration_backup_valid(self, image_default_patch_configuration_backup):
-        return True
+    def __get_current_auto_os_patch_state_for_yum_cron(self):
+        """ Gets current auto OS update patch state for yum-cron """
+        self.composite_logger.log_debug("Fetching current automatic OS patch state in yum-cron service. This includes checks on whether the service is installed, current auto patch enable state and whether it is set to enable on reboot")
+        self.__init_auto_update_for_yum_cron()
+        is_service_installed, enable_on_reboot_value, download_updates_value, apply_updates_value = self.__get_current_auto_os_updates_setting_on_machine()
+
+        apply_updates = self.__get_extension_standard_value_for_apply_updates(apply_updates_value)
+
+        if apply_updates == self.apply_updates_enabled or enable_on_reboot_value:
+            return Constants.AutomaticOSPatchStates.ENABLED
+        # OS patch state is considered to be disabled: a) if it was successfully disabled or b) if the service is not installed
+        elif not is_service_installed or (apply_updates == self.apply_updates_disabled and not enable_on_reboot_value):
```

Comment:
`enable_on_reboot_value` will always be False here

Reply:
Same with the two other instances below

Reply:
yes that is the check.

service is not installed or updates are disabled and enable on reboot is set to False, only then do we consider patch state to be disabled